File: backend/app/schemas/moderator.py
"""Schemas for moderator guidance flow."""

from __future__ import annotations

from typing import List, Literal, Optional

from pydantic import BaseModel, Field

from app.schemas.sessions import ChecklistKey

Actor = Literal["agent", "customer"]
ToneLabel = Literal["negative", "neutral", "positive"]


class TranscriptSegment(BaseModel):
    actor: Actor
    text: str
    timestamp: str


class ModeratorGuidanceRequest(BaseModel):
    session_id: str = Field(..., alias="session_id")
    transcript: List[TranscriptSegment]


class ModeratorGuidanceResponse(BaseModel):
    guidance_id: str
    guidance_text: str
    missing_items: List[ChecklistKey]
    tone_alert: Optional[ToneLabel]
    next_poll_seconds: int | None = None


__all__ = [
    "ModeratorGuidanceRequest",
    "ModeratorGuidanceResponse",
    "TranscriptSegment",
    "ToneLabel",
]

File: frontend/src/types/moderator.ts
import type { ChecklistKey } from "../utils/surveyConfig";

export interface ModeratorTranscriptSegment {
  actor: "agent" | "customer";
  text: string;
  timestamp: string;
}

export interface ModeratorGuidanceResponse {
  guidance_id: string;
  guidance_text: string;
  missing_items: ChecklistKey[];
  tone_alert: "negative" | "neutral" | "positive" | null;
  next_poll_seconds?: number | null;
}
